src/proto/network/pxe.rs

```diff
@@ -0,0 +1,1236 @@
+    pub fn set_parameters(
+        &mut self,
+        new_auto_arp: Option<bool>,
+        new_send_guid: Option<bool>,
+        new_ttl: Option<u8>,
+        new_tos: Option<u8>,
+        new_make_callback: Option<bool>,
+    ) -> Result {
+        (self.set_parameters)(
+            self,
+            new_auto_arp.as_ref(),
+            new_send_guid.as_ref(),
+            new_ttl.as_ref(),
+            new_tos.as_ref(),
+            new_make_callback.as_ref(),
+        )
+        .into()
+    }
+
+    /// Updates the station IP address and/or subnet mask values of a network
+    /// device.
+    pub fn set_station_ip(
+        &mut self,
+        new_station_ip: Option<&IpAddress>,
+        new_subnet_mask: Option<&IpAddress>,
+    ) -> Result {
+        (self.set_station_ip)(self, new_station_ip, new_subnet_mask).into()
+    }
+
+    /// Updates the contents of the cached DHCP and Discover packets.
+    #[allow(clippy::too_many_arguments)]
+    pub fn set_packets(
+        &mut self,
+        new_dhcp_discover_valid: Option<bool>,
+        new_dhcp_ack_received: Option<bool>,
+        new_proxy_offer_received: Option<bool>,
+        new_pxe_discover_valid: Option<bool>,
+        new_pxe_reply_received: Option<bool>,
+        new_pxe_bis_reply_received: Option<bool>,
+        new_dhcp_discover: Option<&Packet>,
+        new_dhcp_ack: Option<&Packet>,
+        new_proxy_offer: Option<&Packet>,
+        new_pxe_discover: Option<&Packet>,
+        new_pxe_reply: Option<&Packet>,
+        new_pxe_bis_reply: Option<&Packet>,
+    ) -> Result {
+        (self.set_packets)(
+            self,
+            new_dhcp_discover_valid.as_ref(),
+            new_dhcp_ack_received.as_ref(),
+            new_proxy_offer_received.as_ref(),
+            new_pxe_discover_valid.as_ref(),
+            new_pxe_reply_received.as_ref(),
+            new_pxe_bis_reply_received.as_ref(),
+            new_dhcp_discover,
+            new_dhcp_ack,
+            new_proxy_offer,
+            new_pxe_discover,
+            new_pxe_reply,
+            new_pxe_bis_reply,
+        )
+        .into()
+    }
+
+    /// Returns a reference to the `Mode` struct.
+    pub fn mode(&self) -> &Mode {
+        unsafe { &*self.mode }
+    }
+}
+
+// EFI_PXE_BASE_CODE_BOOT_*
+#[repr(u16)]
+#[allow(missing_docs)]
+pub enum BootstrapType {
+    Bootstrap = 0,
+    MsWinntRis = 1,
+    IntelLcm = 2,
+    DosUndi = 3,
+    NecEsmpro = 4,
+    IbmWsoD = 5,
+    IbmLccm = 6,
+    CaUnicenterTng = 7,
+    HpOpenview = 8,
+    Altiris9 = 9,
+    Altiris10 = 10,
+    Altiris11 = 11,
+    // NOT_USED_12 = 12,
+    RedhatInstall = 13,
+    RedhatBoot = 14,
+    Rembo = 15,
+    Beoboot = 16,
+    //
+    // Values 17 through 32767 are reserved.
+    // Values 32768 through 65279 are for vendor use.
+    // Values 65280 through 65534 are reserved.
+    //
+    PxeTest = 65535,
+}
+
+/// This struct contains optional parameters for [`BaseCode::discover`].
+// EFI_PXE_BASE_CODE_DISCOVER_INFO
```

Comment:
Do we want to leave these comments around or replace them with doc comments mentioning this is the corresponding C API type?

Reply:
I wasn't planning on making these into proper doc comments, but I'd be fine with just adding something like this to all types:
```rust
/// Corresponds to `$name` type in the C API.
```

Reply:
That would be great, thank you!